sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -117,7 +117,7 @@ object DataType {
   private val FIXED_DECIMAL = """decimal\(\s*(\d+)\s*,\s*(\-?\d+)\s*\)""".r
   private val CHAR_TYPE = """char\(\s*(\d+)\s*\)""".r
   private val VARCHAR_TYPE = """varchar\(\s*(\d+)\s*\)""".r
-  private val COLLATED_STRING_TYPE = """string\s+COLLATE\s+'([\w_]+)'""".r
+  private val COLLATED_STRING_TYPE = """string\s+COLLATE\s+([\w_]+)""".r
```

Comment:
When you define `collationName=identifier` where `identifier` can be:
```
quotedIdentifier
    : BACKQUOTED_IDENTIFIER
    | {double_quoted_identifiers}? DOUBLEQUOTED_STRING
    ;
``` 
but `\w` captures only `[a-zA-Z_0-9]`. Is it in purpose, and what happens if an user pass the collation name as
```
`UCS_BASIC_LCASE` or
"UCS_BASIC_LCASE"
```

Reply:
Nice catch! Since `collate_key_word_as_identifier` is false I guess we only need to support backticks. I added the tests for them.